tools/pubsys/src/aws/client.rs

```diff
@@ -1,159 +1,169 @@
-use async_trait::async_trait;
-use pubsys_config::AwsConfig;
-use rusoto_core::{request::DispatchSignedRequest, HttpClient, Region};
-use rusoto_credential::{
-    AutoRefreshingProvider, AwsCredentials, CredentialsError, DefaultCredentialsProvider,
-    ProfileProvider, ProvideAwsCredentials,
-};
-use rusoto_ebs::EbsClient;
-use rusoto_ec2::Ec2Client;
-use rusoto_ssm::SsmClient;
-use rusoto_sts::{StsAssumeRoleSessionCredentialsProvider, StsClient};
-use snafu::ResultExt;
-
-pub(crate) trait NewWith {
-    fn new_with<P, D>(request_dispatcher: D, credentials_provider: P, region: Region) -> Self
-    where
-        P: ProvideAwsCredentials + Send + Sync + 'static,
-        D: DispatchSignedRequest + Send + Sync + 'static;
-}
+use aws_config::profile::ProfileFileCredentialsProvider;
 
-impl NewWith for EbsClient {
-    fn new_with<P, D>(request_dispatcher: D, credentials_provider: P, region: Region) -> Self
-    where
-        P: ProvideAwsCredentials + Send + Sync + 'static,
-        D: DispatchSignedRequest + Send + Sync + 'static,
-    {
-        Self::new_with(request_dispatcher, credentials_provider, region)
-    }
-}
+use aws_config::{ConfigLoader, SdkConfig};
+use aws_sdk_ec2::{Endpoint, Region};
```

Comment:
nit: maybe we should use `aws_sdk_sts::Region` here, since that's the only service we end up talking to.